lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -497,6 +502,18 @@ instance {-# OVERLAPS #-} Arbitrary (Key 'RootK XPrv, Passphrase "encryption")
         let key = generateKeyFromSeed (seed, mempty) pwd
         return (key, pwd)
 
+instance Arbitrary SlotId where
+    shrink _ = []
+    arbitrary = SlotId <$> arbitrary <*> arbitrary
+
+instance Arbitrary SortOrder where
+    shrink _ = []
+    arbitrary = genericArbitrary
+
+instance Arbitrary UTCTime where
+    shrink _ = []
+    arbitrary = posixSecondsToUTCTime . toEnum <$> arbitrary
```

Comment:
In other places, we generate arbitrary `UTCTime` values with:
```
lib/core/test/unit/Data/Time/TextSpec.hs
29:import Test.QuickCheck.Instances.Time

lib/core/test/unit/Cardano/Wallet/DBSpec.hs
143:import Test.QuickCheck.Instances.Time

lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs
177:import Test.QuickCheck.Instances.Time
```
But perhaps your instance is nicer. Perhaps we should use the same instance in all places?

Reply:
🤷‍♂ The `Test.QuickCheck.Instances.Time` seem to catch removing one of the bound-checks in `Sqlite.hs` — i.e. work fine — so I'll go with that for consistency.

Reply:
Great!

Having said that, I seem to remember that the `Test.QuickCheck.Instances.Time` instance generates dates that are biased toward a very long time ago. I think it would be worth investigating the distribution, and seeing if we need to improve on it.

Though that work is definitely outside the scope of this PR. I've added an item to the blackboard here: https://github.com/input-output-hk/cardano-wallet/wiki/Blackboard#investigate-the-arbitrary-instance-for-utctime-in-testquickcheckinstancestime